module/VuFind/src/VuFind/Net/IpAddressUtils.php

```diff
@@ -124,4 +124,34 @@ public function isInRange($ip, $ranges)
         }
         return false;
     }
+
+    /**
+     * Truncate an IP address to the given number of IPv4 octets
+     * or IPv6 hextets, depending what kind of IP address it is.
+     *
+     * @param string $ip IP address to truncate
+     * @param int $ipv4Octets Number of octets to return if it is IPv4
+     * @param int $ipv6Hextets Number of hextets to return if it is IPv6
+     *  
+     */
+    public function truncate($ip, $ipv4Octets = null, $ipv6Hextets = null)
```

Comment:
I get that in theory this logic could be based on actual subnets, not a simple truncation of the address.  This seems much easier though to understand and configure.

Reply:
Agreeed. We can add subnet handling later if needed.